pkg/controlplane/apiserver/config.go

```diff
@@ -74,6 +74,9 @@ type Extra struct {
 	StorageFactory          serverstorage.StorageFactory
 	EventTTL                time.Duration
 
+	// FIXME: Does it belong here or genericapiserver?
+	StorageInitializationTimeout time.Duration
```

Comment:
@jpbetz - thoughts?

Reply:
I think so.  @sttts can you confirm?  Its seems generally useful.

Reply:
Just for clarity, as we are before doing the big generic controlplane package renaming/movement, this is the generic controlplane side at the moment.

To answer the question: would an aggregated apiserver benefit from/need this logic, then k8s.io/apiserver would be the right place.

Reply:
From looking at the code above, this block:
```golang
for version, storageMap := range apiGroupInfo.VersionedResourcesStorageMap {
			for resource, storage := range storageMap {
				if withReadiness, ok := storage.(rest.StorageWithReadiness); ok {
					gvr := metav1.GroupVersionResource{
						Group:    groupName,
						Version:  version,
						Resource: resource,
					}
					storageReadinessHook.RegisterStorage(gvr, withReadiness)
				}
			}
		}
```
could easily move into `GenericAPIServer.Install(Legacy)APIGroups`. This seems to indicate that it should be a generic functionality of GenericAPIServer.

Reply:
Another hint: what about kube-aggregator and apiextensions-apiserver? They also need this.

Reply:
From the KEP:
"Finally, we suggest starting just with builtin resources, as CRDs can be created at any time, making it harder to implement with unclear gain. If it appears to not be enough (or alternatively we realize that only a subset of builtin resource types is needed), this decision can easily be revisited in the future."

while we may want to change that later, for beta we want to start only with builtin resources.

Reply:
That said, I will look into moving this logic to GenericAPIServer.